doc/manuals/name-encoding.md

```diff
@@ -0,0 +1,47 @@
+# Database and collection name encoding and decoding
+
+MongoDB imposes a set of restrictions regarding the characters which cannot be used as part of the databases and collection names, as well as other limitations. These restrictions are detailed at https://docs.mongodb.com/manual/reference/limits/
+
+Since the names of the databases and the collections used by the STH component are composed from the concatenation of the values assigned to the service, service path, entity id, entity type, attribute name and attribute type (depending on the chosen data model) and we do not want the potential users of the STH component to inherit the restrictions imposed by MongoDB (as the database used by the STH component), a codification mechanism is used before conforming the name of the databases and collections.
+
+The codification mechanism is detailed next:
+    1. For database names:
+        1. Encode the forbidden characters (i.e., `/``\``.``(blank space)``"``$``(null character)`) using an escaping character (`x`) and the numerical Unicode code for each character. For instance, the `$` character will be encoded as `x0024`.
+        2. User defined strings already using the above encoding must be escaped prepending another `x`, for instance: the text `x002a` will be encoded as `xx002a`.
+        3. To avoid the case sensitivity restrictions imposed by MongoDB to the database names, uppercase characters (since they are supposed to be less common than lowercase ones) are also be encoded (i.e., `Service` -> `x0053ervice`).
+        4. The name separator character (`xffff`) is not decoded.
```

Comment:
decoded or encoded?

In the case it is 'encoded', what about user defined strings containing the `xffff` substring?